### ISSUE TITLE
Do we care about k-fold cross validation for large datasets?

### ISSUE THREAD

**Issue by mrocklin** [dask/dask-ml#303]
When considering incremental models how much do we care about k-fold cross validation?  My understanding is that the k-fold bit is an effort to train on all of our data, while still protecting testing data for each fold.  I wonder how often using all of the data is a requirement in the cases where `Incremental` will be used.  My guess would be that this becomes less important.

**Comment by mrocklin** [comment]
Similarly for normal 1-fold cross validation what should we do when the user doesn't provide testing data.  I can imagine a couple options:

1.  Randomly perform a train/test split
2.  Select the first or last block for testing data

The first is safer but probably slows down single-machine out-of-core computing cases when the data doesn't sit in memory (we would have to read through all of it).  The second is much faster in these cases, but assumes uniformity of the data.  Probably we should provide both options.  I'm curious what a default should be.

This also came up in https://github.com/dask/dask-ml/issues/264#issuecomment-404490151

cc @ogrisel, your intuition would be valuable here.

**Comment by shoyer** [comment]
No, this is usually overkill. For large datasets, the usual pattern is to divide them into three pieces:
- train: for direct model optimization
- tune: for hyper-parameter optimization
- evaluate: for evaluating final models

**Comment by mmccarty** [comment]
FWIW I have a use case for a folding on large data, however I'm defining my own cross validation class. Also, I am able to get a lift in model performance with large data. Perhaps it's worth thinking about generalizing?
